src/VisualStudio/Core/Def/Diagnostics/VisualStudioDiagnosticAnalyzerService.cs

```diff
@@ -306,7 +306,10 @@ public void RunAnalyzers(IVsHierarchy? hierarchy)
             var project = GetProject(hierarchy);
             var solution = _workspace.CurrentSolution;
             var projectOrSolutionName = project?.Name ?? PathUtilities.GetFileName(solution.FilePath);
-            var analysisScope = project != null ? _globalOptions.GetBackgroundAnalysisScope(project.Language) : SolutionCrawlerOptionsStorage.BackgroundAnalysisScopeOption.DefaultValue;
+            var isAnalysisDisabled = project != null
+                ? _globalOptions.IsAnalysisDisabled(project.Language, out _, out _)
+                : SolutionCrawlerOptionsStorage.BackgroundAnalysisScopeOption.DefaultValue == BackgroundAnalysisScope.None &&
+                  SolutionCrawlerOptionsStorage.CompilerDiagnosticsScopeOption.DefaultValue == CompilerDiagnosticsScope.None;
```

Comment:
so confused what it means to have options, but then like.. have fallbacks.  when does that happen?

Reply:
I have cleaned this up to be lot more intuitive. If non-null project is specified, we use it, otherwise we compute these values by looking at all projects in the solution.